account_invoice_constraint_chronology_in_date_range/models/account_move.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2024 Foodles (https://www.foodles.co/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.osv import expression
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _get_older_conflicting_invoices_domain(self):
+        self.ensure_one()
+        if not self.journal_id.sequence_id.use_date_range:
+            return super()._get_older_conflicting_invoices_domain()
+        else:
+            date_range_id = self.env["ir.sequence.date_range"].search(
+                [
+                    ("sequence_id", "=", self.journal_id.sequence_id.id),
+                    ("date_from", "<=", self.invoice_date),
+                    ("date_to", ">=", self.invoice_date),
+                ],
+                limit=1,
+            )
```

Comment:
**todo:** This can be simplified to the following (and it even creates the `ir.sequence.date_range` if need be):

```suggestion
            date_range_id = self.journal_id.sequence_id._get_current_sequence(self.invoice_date)
```

Same in `_get_newer_conflicting_invoices_domain`